csharp/src/Apache.Arrow/Arrays/ArrayData.cs

```diff
@@ -30,10 +30,25 @@ public sealed class ArrayData : IDisposable
         public readonly int Offset;
         public readonly ArrowBuffer[] Buffers;
         public readonly ArrayData[] Children;
+        public readonly ArrayData Dictionary; //Only used for dictionary type
 
         public ArrayData(
             IArrowType dataType,
             int length, int nullCount = 0, int offset = 0,
+            IEnumerable<ArrowBuffer> buffers = null, IEnumerable<ArrayData> children = null) :
+            this(dataType, null, length, nullCount, offset, buffers, children)
+        { }
+
+        public ArrayData(
+            IArrowType dataType,
+            int length, int nullCount = 0, int offset = 0,
+            ArrowBuffer[] buffers = null, ArrayData[] children = null) :
+            this(dataType, null, length, nullCount, offset, buffers, children)
+        { }
+
+        public ArrayData(
+            IArrowType dataType, ArrayData dictionary,
```

Comment:
To avoid the ambiguity between ctors, `dictionary` is positioned the second argument...

Reply:
I noticed that changing original ctor's optional parameters to normal parameters can avoid the ambiguity and keep backward binary compatibility.
Please excuse my ignorance...


```
        public ArrayData(
            IArrowType dataType,
            int length, int nullCount, int offset,
            IEnumerable<ArrowBuffer> buffers, IEnumerable<ArrayData> children) :
            this(dataType, length, nullCount, offset, buffers, children, null)
        { }

        public ArrayData(
            IArrowType dataType,
            int length, int nullCount = 0, int offset = 0,
            IEnumerable<ArrowBuffer> buffers = null, IEnumerable<ArrayData> children = null, ArrayData dictionary = null)
        {
           ...
        }
```